### DISCUSSION TITLE
OpenMsgStore MAPI_E_FAILONEPROVIDER

### DISCUSSION THREAD

**Discussion by ulued** [microsoft/mfcmapi#461] [Q&A]
I've derived from your code the usage of OpenMsgStore, using it in a service (session id 0) running in a user context.

Outlook is Outlook 365 64bit, 16.0.11727.20222
 
Everything is working fine on accessing an OnPrem Exchange Server.

Accessing Exchange 365 always gives MAPI_E_FAILONEPROVIDER even on opening the sane message store as in the Outlook profile.

I've another executable using the same code and when started in the same user context (e.g. by double click, session id 1) E365 access is OK. The same behavior as in the service occurs using the executable in the task planner.

You have already mentioned an adal bug in Outlook 365 >1806 and recommended to downgrade, which is not an option for me.

Any suggestion to what might be going wrong and a workaround/fix for it would be appreciated.

In MfcMapi open other mailboxes from GAL does not work either. It opens the other mailbox, but only displays PeoplePublicData - no error

**Comment by ulued** [comment]
I solved the issue for my service by disabling the "Advanced Security Configuration for IE" in the Server Mangager.

As for MCMAPI it always return MAPI_E_FAILONEPROVIDER on trying to connect to a E365 mailbox.

![MFCMAPI_MAPI_FAIL_ONEPROVIDER](https://user-images.githubusercontent.com/41414446/82290762-f0659a00-99a7-11ea-9678-44fcfc4f61ca.png)

**Comment by ulued** [comment]
Version used :MFCMAPI 20.0.20110.01

**Comment by ulued** [comment]
Error persists - I could only use OL2016 without updates. Updating to current patchlevel raises the error.

**Comment by leethobbit** [comment]
I have Outlook for Office 365 MSO (16.0.11929.20708) 32-bit and am experiencing the same issues trying to connect to my accounts.
